scilpy/dwi/operations.py

```diff
@@ -119,104 +122,168 @@ def compute_dwi_attenuation(dwi_weights: np.ndarray, b0: np.ndarray):
     dwi_attenuation : np.ndarray
         Signal attenuation (Diffusion weights normalized by the B0).
     """
+    # Avoid division by 0. Remember coordinates where the b0 was 0. We will set
+    # those voxels to 0 in the final result.
+    zeros_mask = b0 == 0
+
     b0 = b0[..., None]  # Easier to work if it is a 4D array.
 
-    # Make sure that, in every voxels, weights are lower in the b0. Should
-    # always be the case, but with the noise we never know!
+    # Make sure that, in every voxel, weights are lower in the dwi than in the
+    # b0. Should always be the case, but with the noise we never know!
     erroneous_voxels = np.any(dwi_weights > b0, axis=3)
     nb_erroneous_voxels = np.sum(erroneous_voxels)
     if nb_erroneous_voxels != 0:
         logging.info("# of voxels where `dwi_signal > b0` in any direction: "
-                     "{}".format(nb_erroneous_voxels))
+                     "{}. They were set to the b0 value to allow computing "
+                     "signal attenuation."
+                     .format(nb_erroneous_voxels))
         dwi_weights = np.minimum(dwi_weights, b0)
 
     # Compute attenuation
+    b0[zeros_mask] = 1e-6
     dwi_attenuation = dwi_weights / b0
-
-    # Make sure we didn't divide by 0.
-    dwi_attenuation[np.logical_not(np.isfinite(dwi_attenuation))] = 0.
+    dwi_attenuation *= ~zeros_mask[:, :, :, None]
 
     return dwi_attenuation
 
 
 def detect_volume_outliers(data, bvecs, bvals, std_scale,
```

Comment:
Could we change the order of the parameters here? We tend to give data, bvals, bvecs usually? (Don't forget to also change it in the docstring below)